autoprotocol/compound.py

```diff
@@ -7,20 +7,58 @@ class Compound:
 
     Parameters
     ----------
-    InChI : Standard International Chemical Identifier
+    comp_format : Compound string input format
+    value : Compound string
 
     """
 
-    def __init__(self, InChI):
-        if self.is_valid(InChI):
-            self.InChI = InChI
+    def __init__(self, comp_format, value):
+        compound_formats = ["Daylight Canonical SMILES", "InChI"]
+        if comp_format in compound_formats:
+            self.format = comp_format
         else:
-            raise CompoundError(f"{InChI} is not a valid InChI key")
+            raise CompoundError(f"{comp_format} is not an acceptable Compound format.")
 
-    @staticmethod
-    def is_valid(InChI):
-        pattern = r"^InChI\=1S?\/[^\s]+(\s|$)"
-        return re.match(pattern, InChI)
+        if self.is_valid(value):
+            self.value = value
+        else:
+            raise CompoundError(f"{value} is not a valid {self.format} value.")
+
+    def is_valid(self, compound):
+        inchi_pattern = r"^InChI\=1S?\/[^\s]+(\s|$)"
+        non_inchi_pattern = r"^(?!InChI=)\w\S+$"
+
+        if self.format == "InChI":
+            return bool(re.match(inchi_pattern, compound))
+        elif self.format == "Daylight Canonical SMILES":
+            return bool(
+                re.match(non_inchi_pattern, compound)
+                and not re.search("[~?!$%^&;'Jj]", compound)
```

Comment:
Can you explain `re.search("[~?!$%^&;'Jj]"`? Perhaps add a comment and make `"[~?!$%^&;'Jj]"` a variable explaining what the regex is matching against.

If we're not 100% confident and clear on these regexs, we should remove this validation altogether.

Reply:
Added comment. There are symbols that are not used in SMILES, but are used in other formats like SMART. Wanted to include these here in case we add other formats in the future.